src/controller/CHIPDeviceControllerSystemState.h

```diff
@@ -190,7 +193,7 @@ class DeviceControllerSystemState
             mUnsolicitedStatusHandler != nullptr && mExchangeMgr != nullptr && mMessageCounterManager != nullptr &&
             mFabrics != nullptr && mCASESessionManager != nullptr && mSessionSetupPool != nullptr && mCASEClientPool != nullptr &&
             mGroupDataProvider != nullptr && mReportScheduler != nullptr && mTimerDelegate != nullptr &&
-            mSessionKeystore != nullptr && mSessionResumptionStorage != nullptr;
+            mSessionKeystore != nullptr && mSessionResumptionStorage != nullptr && mBDXTransferServer != nullptr;
```

Comment:
Why is it mandatory to provide a BDX transfer server?

Reply:
Is the idea here that it's internal SDK code that always provides this, not API consumers?

Reply:
At the moment this is the idea. I believe it can be changed in a followup if needed.